### DISCUSSION TITLE
Canvas Layers Order Issue using IText and Text

### DISCUSSION THREAD

**Discussion by akalana** [fabricjs/fabric.js#6862] [Q&A]
We are using an array to store Canvas elements and we iterate the array however when we push this to the canvas **IText and Text** goes to the bottom layer in the Canvas.
Does anyone know a solution for this or have you run into a similar issue in the past? Thank you

**Comment by asturur** [comment]
Hi @akalana your description is very incomplete for someone to understand what are you doing.
Can you reproduce this in a jsfiddle example, cutting out the irrelevant parts to the problem?
If someone sees it can maybe help.
